File: backend/src/main/java/com/etchedjournal/etched/models/jooq/converter/IdConverter.java
package com.etchedjournal.etched.models.jooq.converter;

import com.etchedjournal.etched.utils.id.IdSerializer;
import org.jooq.Converter;

public class IdConverter implements Converter<Long, String> {

    @Override
    public String from(Long databaseObject) {
        return IdSerializer.INSTANCE.serialize(databaseObject);
    }

    @Override
    public Long to(String userObject) {
        return IdSerializer.INSTANCE.deserialize(userObject);
    }

    @Override
    public Class<Long> fromType() {
        return Long.class;
    }

    @Override
    public Class<String> toType() {
        return String.class;
    }
}

File: backend/src/main/kotlin/com/etchedjournal/etched/models/entity/IdConverter.kt
package com.etchedjournal.etched.models.entity

import com.etchedjournal.etched.utils.id.IdSerializer
import javax.persistence.AttributeConverter
import javax.persistence.Converter

/**
 * Converts ids between database long representation to a user friendly string
 */
@Converter
class IdConverter : AttributeConverter<String, Long> {
    override fun convertToDatabaseColumn(attribute: String): Long {
        return IdSerializer.deserialize(attribute)
    }

    override fun convertToEntityAttribute(dbData: Long): String {
        return IdSerializer.serialize(dbData)
    }
}
